fastestimator/backend/tensor_normalize.py

```diff
@@ -0,0 +1,123 @@
+    ```python
+    n = np.array([[[1.0, 2.0], [3.0, 4.0]], [[5.0, 6.0], [7.0, 8.0]]])
+    b = fe.backend.tensor_normalize(n, None, None)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, 4.5, 2.29128784747792)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, (4., 5.), (2.23606798, 2.23606798))  # [[[-1.34164073, -1.34164073], [-0.44721358, -0.44721358]], [[ 0.44721358,  0.44721358], [ 1.34164073,  1.34164073]]]
+    ```
+
+    This method can be used with TensorFlow tensors:
+    ```python
+    t = tf.constant([[[1.0, 2.0], [3.0, 4.0]], [[5.0, 6.0], [7.0, 8.0]]])
+    b = fe.backend.tensor_normalize(n, None, None)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, 4.5, 2.29128784747792)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, (4., 5.), (2.23606798, 2.23606798))  # [[[-1.34164073, -1.34164073], [-0.44721358, -0.44721358]], [[ 0.44721358,  0.44721358], [ 1.34164073,  1.34164073]]]
+    ```
+
+    This method can be used with PyTorch tensors:
+    ```python
+    p = torch.tensor([[[1.0, 2.0], [3.0, 4.0]], [[5.0, 6.0], [7.0, 8.0]]])
+    b = fe.backend.tensor_normalize(n, None, None)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, 4.5, 2.29128784747792)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, (4., 5.), (2.23606798, 2.23606798))  # [[[-1.34164073, -1.34164073], [-0.44721358, -0.44721358]], [[ 0.44721358,  0.44721358], [ 1.34164073,  1.34164073]]]
+    ```
+
+    Args:
+        tensor: The input value.
+        mean: The mean which needs to applied(eg: None, 3.8, (1.9, 2.0, 2.9))
+        std: The standard deviation which needs to applied(eg: None, 3.8, (1.9, 2.0, 2.9))
+
+    Returns:
+        The normalized values of `tensor`.
+
+    Raises:
+        ValueError: If `tensor` is an unacceptable data type.
+    """
+    channel_level = False
+    if isinstance(mean, Tuple) or isinstance(std, Tuple):
+        channel_level = True
+
+    mean = get_mean(tensor, mean, channel_level)
+    std = get_std(tensor, std, channel_level)
+    epsilon = get_epsilon(tensor, epsilon)
+
+    if tf.is_tensor(tensor) or isinstance(tensor, torch.Tensor) or isinstance(tensor, np.ndarray):
+        tensor = (tensor - mean) / (std + epsilon)
+
+        # for testing purpose we are not changing the channels
+        #if isinstance(tensor, torch.Tensor):
```

Comment:
why is this still here?

Reply:
@TortoiseHam  current assumption is both the tensorflow and torch would provide tensor to be channel last, but torch is channel first operation, than this needs to be updated.

Reply:
torch should be assumed to be channel first